@here/harp-examples/example-browser.ts

```diff
@@ -50,22 +50,76 @@ function exampleBrowser(exampleDefinitions: ExampleDefinitions) {
         Object.keys(exampleDefinitions)
             .sort()
             .forEach(pageUrl => {
-                const linkName = getName(pageUrl);
+                const linkName = getName(pageUrl).replace(new RegExp("-", "g"), " ");
+                const linkElements = linkName.split(" / ");
+                // Create category 'miscellaneous' for examples without category name.
+                let linkSubMenu = linkElements.length > 1 ? linkElements[0] : "miscellaneous";
+
+                // Style: uppercase the first letter in case CSS lets lowercase.
+                linkSubMenu = linkSubMenu.charAt(0).toUpperCase() + linkSubMenu.slice(1);
+                // Create category if needed.
+                if (categories[linkSubMenu] === undefined) {
+                    categories[linkSubMenu] = [];
+                }
+                const visibleText =
+                    linkElements.length === 1 ? linkName : linkElements.slice(1).join(" / ");
                 const linkElement = createDomElement<HTMLAnchorElement>("a", {
-                    innerText: linkName,
+                    innerText: visibleText,
                     href: "#" + pageUrl,
                     className: "link"
                 });
-                menuElements.push(linkElement);
-                exampleListElement.appendChild(linkElement);
+                (linkElement as any).nameWithCategory = linkName;
+                (linkElement as any).nameWithoutCategory = visibleText;
+                categories[linkSubMenu].push(linkElement);
+                elements.push(linkElement);
+            });
+
+        Object.keys(categories)
+            .sort()
+            .forEach(category => {
+                if (!orderedCategories.includes(category)) {
+                    orderedCategories.push(category);
+                }
+            });
+
+        orderedCategories.forEach(menuElement => {
+            const menu = document.createElement("div");
+            const title = document.createElement("h2");
+            title.innerHTML = menuElement;
+            titleElements.push(title);
+            menu.appendChild(title);
+            categories[menuElement].forEach(anchor => {
+                menu.appendChild(anchor);
             });
+            exampleListElement.appendChild(menu);
+        });
     }
 
-    function installFilter(filterInput: HTMLInputElement, fileteredElements: HTMLElement[]) {
+    function installFilter(filterInput: HTMLInputElement) {
         filterInput.addEventListener("input", e => {
             const filterValue = (filterInput.value || "").trim();
 
-            for (const element of fileteredElements) {
+            if (filterValue) {
```

Comment:
The whole if-else block should be called only when actual condition changes, in this case ```filterValue.length > 0```, and not on every keypress. Also remember about truthy values.